### ISSUE TITLE
Incorrect output for a C++ type containing a function pointer

### ISSUE THREAD

**Issue by erankavija** [doxygen/doxygen#10703]
**Describe the bug**
The file
```
#include <vector>

class A;

/**
 * @brief Some class
 */
class C {
public:
  /// Some object 1
  std::vector<std::vector<A> (*)(A)>
      someObject1;
};
```

produces 

`
<type>std::vector&lt; std::vector&lt; A &gt; A</type>
`

in the xml for the type of someObject1.

**Expected behavior**
The type should be parsed and output correctly.

**To Reproduce**
[doxygen_issue.tar.gz](https://github.com/doxygen/doxygen/files/14452632/doxygen_issue.tar.gz)

**Version**
OS:
```
LSB Version:    n/a
Distributor ID: Arch
Description:    Arch Linux
Release:        rolling
Codename:       n/a
```
Doxygen: compiled from sources
`1.11.0 (70ca40726017f149f050f3448e7258bd504a81d6)`

**Comment by albert-github** [comment]
I've just pushed a proposed patch, pull request #10705

**Comment by albert-github** [comment]
Code has been integrated in master on GitHub (please don't close the issue as this will be done at the moment of an official release).

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.11.0.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
